### ISSUE TITLE
Accessibility tools cannot focus the form contained in control correctly

### ISSUE THREAD

**Issue by Olina-Zhang** [dotnet/winforms#7245]
### .NET version

7.0.100-preview.6.22281.2

### Did it work in .NET Framework?

Yes

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

No, this issue still repro in .Net Core 3.1/5.0/6.0

### Issue description

Accessibility tools cannot focus the form contained in control correctly

**.Net 7.0:** repro this issue
![image](https://user-images.githubusercontent.com/26474449/171356598-169c8d7d-6844-41de-87a6-eb3db53645e7.png)

**.Net framework 4.8:** not repro this issue
![image](https://user-images.githubusercontent.com/26474449/171357191-93058d54-7c10-4820-8533-0da9c911148b.png)

### Steps to reproduce

1.	Create a Winforms .net core application
2.	Add a panel control to Form1
3.	Add a new form: Form2 to this application
4.	Double-click Form1, add following code:
            
            Form2 form2 = new Form2();
            form2.TopLevel = false;
            form2.Dock = DockStyle.Fill;
            form2.Parent = panel1;
            form2.Show();

5.	Build and run app
6.	Launch Accessibility tools: AccessibilityInsight/Inspect/Narrator to focus on Form2

or using attached application:
[WinFormsApp16.zip](https://github.com/dotnet/winforms/files/8812252/WinFormsApp16.zip)

**Comment by merriemcgaw** [comment]
Can you confirm if this works for controls inside of containers? I think it would but we want to be sure.

**Comment by Olina-Zhang** [comment]
> Can you confirm if this works for controls inside of containers? I think it would but we want to be sure.

it works well when calling form outside of containers, accessibility tools can focus on it with right position.

**Comment by vladimir-krestov** [comment]
The reason of the regression is #2096, when BoundingRectangle was changed.

**Comment by Cassie-Li01** [comment]
Verified on latest .NET core 7.0 build: .NET 7.0.100-preview.7.22330.8, issue is fixed. Screenshot as below:
![image](https://user-images.githubusercontent.com/96758262/176831714-08c0042c-5392-451b-8270-328a3c6ed3e6.png)
![image](https://user-images.githubusercontent.com/96758262/176831794-b4dc6ffb-849f-4be9-8db8-14baf3793a3c.png)
![image](https://user-images.githubusercontent.com/96758262/176831885-210e4d8d-cb4e-4a39-b732-8e1444a747a7.png)

**Comment by Amy-Li03** [comment]
Verified on .NET 7.0 test pass build: .NET 7.0.100-preview.7.22370.3, issue was fixed.
